src/netbase.cpp

```diff
@@ -48,36 +48,59 @@ std::vector<CNetAddr> WrappedGetAddrInfo(const std::string& name, bool allow_loo
     ai_hint.ai_protocol = IPPROTO_TCP;
     // We don't care which address family (IPv4 or IPv6) is returned
     ai_hint.ai_family = AF_UNSPEC;
+
     // If we allow lookups of hostnames, use the AI_ADDRCONFIG flag to only
     // return addresses whose family we have an address configured for.
     //
     // If we don't allow lookups, then use the AI_NUMERICHOST flag for
     // getaddrinfo to only decode numerical network addresses and suppress
     // hostname lookups.
-    ai_hint.ai_flags = allow_lookup ? AI_ADDRCONFIG : AI_NUMERICHOST;
+    const int ai_flags = allow_lookup ? AI_ADDRCONFIG : AI_NUMERICHOST;
+
+    // getaddrinfo is called twice and to prevent duplicates a set is used
+    // std::set used over a std::unordered_set as CNetAddr does not have a hash function
+    std::set<CNetAddr> resolved_addresses_set;
+
+    auto callgetaddrinfo = [&](const int flags, const bool only_add_local_addr) {
```

Comment:
nit: function arguments that are passed by value need not be `const`
```suggestion
    auto callgetaddrinfo = [&](int flags, bool only_add_local_addr) {
```
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in